### DISCUSSION TITLE
Flow Rate not storing

### DISCUSSION THREAD

**Discussion by Akujiproxy** [Jyers/Marlin#1131] [Q&A] [answered]
Anybody else having the issue that the Flow Rate setting doesn't seem to be stored. After restarting the printer, it will be reverted back to 100.

**Comment by Cal-LF** [comment]
Hi, the flow rate is used as a one time change to your current print. The flow rate is a percentage of that set in the slicer much the same as the speed setting. It is not a permanent setting that will be saved to eprom.
A search of the term "flow" will produce a reference to #756.

**Reply by Akujiproxy** [reply]
I wonder why it wouldn't be stored as the change is permanently required

**Reply by Cal-LF** [reply]
Any permanent change can be made in the slicer.

**Reply by Akujiproxy** [reply]
Yeah I realize that but it would be more practical in the firmware.

**Reply by Cal-LF** [reply]
Refer to the #756. This is just how marlin handles it.
I would not want the flow setting saved permanently.

**Reply by Akujiproxy** [reply]
Yeah i read it. May I ask why you don't think its a good idea?

**Comment by Jyers** [comment, marked as answer]
It's not really a question of whether or not it's a good idea. I would be fine either way but it's just how marlin deals with it. To make it persistent would require overwriting the way marlin handles the flowrate which is against the whole point of this project.

**Reply by Akujiproxy** [reply]
Yeah that's fair, and its not a big problem for me. I was just curious as to his/her reason for actively not wanting it that way. Just so I can get a better understanding of the pro's and con's of these things.

**Comment by Cal-LF** [comment]
@Akujiproxy My thought about saving the flow setting is if you have your flow at 97 like I do, making the printer setting to something other than 100% is redundant and somewhat confusing to what the flow really is. It is all relative to the estep setting applied at the printer. My estep is set to 97 also.

**Reply by Akujiproxy** [reply]
I'm afraid I still don't quite understand how that matters? Surely you would figure out and set your e-step first and then flow. And the two are related but both wouldn't change over time. Or am I missing something?
